include/bluetooth/gatt.h

```diff
@@ -452,6 +452,58 @@ ssize_t bt_gatt_attr_read_service(struct bt_conn *conn,
 	const Z_STRUCT_SECTION_ITERABLE(bt_gatt_service_static, _name) =\
 						BT_GATT_SERVICE(attr_##_name)
 
+/** @def _BT_GATT_ATTRS_ARRAY_DEFINE
+ *  @brief Statically define attribute array for a service.
+ *
+ *  Helper macro to statically define attribute array for a service.
+ *  It is used in the @ref BT_GATT_SERVICE_INSTANCE_DEFINE macro.
+ *
+ *  @param _n         Item number.
+ *  @param _instances Array of instances to pass as user context to the
+ *                    attribute callbacks.
+ *  @param _attrs_def Macro provided by the user that defines attribute
+ *                    array for the serivce. This macro should accept single
+ *                    parameter which is the instance context.
+ */
+#define _BT_GATT_ATTRS_ARRAY_DEFINE(n, _instances, _attrs_def)	\
+	static struct bt_gatt_attr attrs_##n[] = _attrs_def(_instances[n]);
+
+/** @def _BT_GATT_SERVICE_ARRAY_ITEM
+ *  @brief Single item listed in the service structure array.
+ *
+ *  Helper macro to declare single item listed in the service structure array.
+ *  It is used in the @ref BT_GATT_SERVICE_INSTANCE_DEFINE macro.
+ *
+ *  @param _n Item number.
+ */
+#define _BT_GATT_SERVICE_ARRAY_ITEM(_n, _) BT_GATT_SERVICE(attrs_##_n),
```

Comment:
I guess these one don't really need to be exposed in the documentation, or are you planning on using them elsewhere?

Reply:
No, this is a internal macro that should only be used by BT_GATT_SERVICE_INSTANCE_DEFINE. I can remove the documentation completely